generic3g/specs/FieldSpec.F90

```diff
@@ -168,40 +176,96 @@ subroutine allocate(this, rc)
 
       integer :: status
       type(ESMF_FieldStatus_Flag) :: fstatus
-      
+      integer, allocatable :: final_lbounds(:),final_ubounds(:)
+      integer :: num_levels, total_ungridded_dims
+
+      num_levels = this%vertical_geom%get_num_levels()
+      if (this%vertical_dim == VERTICAL_DIM_NONE) then
+         allocate(final_lbounds,source=this%ungridded_dims%get_lbounds())
+         allocate(final_ubounds,source=this%ungridded_dims%get_ubounds())
+      else
+         total_ungridded_dims = size(this%ungridded_dims%get_lbounds())
+         allocate(final_lbounds(total_ungridded_dims+1)) 
+         allocate(final_ubounds(total_ungridded_dims+1)) 
+         if (this%vertical_dim == VERTICAL_DIM_CENTER) then
+            final_lbounds(1)=1
+            final_lbounds(2:)=this%ungridded_dims%get_lbounds()
+            final_ubounds(1)=num_levels
+            final_ubounds(2:)=this%ungridded_dims%get_ubounds()
+         else if (this%vertical_dim == VERTICAL_DIM_EDGE) then
+            final_lbounds(1)=0
+            final_lbounds(2:)=this%ungridded_dims%get_lbounds()
+            final_ubounds(1)=num_levels
+            final_ubounds(2:)=this%ungridded_dims%get_ubounds()
+         end if
+      end if
```

Comment:
@tclune this requires some discussion, I saw that you had a partial idea of how this could be folded into the ungridded dimension, but when, we can't do this when the component spec is parsed as we don't know the size of the vertical dimension, only if it is none/center/edge. So we only can modify the ungridded dimension in a spec after we have the actual vertical geometry. So where do we do this? In the VariableSpec? That's the first place we have the information the variable specs ungridded dim object could be modified if the variable indeed has a vertical coordinate. 

I had actually thought about doing this.

Reply:
I agree it needs some thought.   I'll try to clear my head in the morning to think about this.